horreum-backend/src/main/java/io/hyperfoil/tools/horreum/datastore/CollectorApiDatastore.java

```diff
@@ -93,10 +96,17 @@ private static void verifyPayload(ObjectMapper mapper, CollectorApiDatastoreConf
         // Verify that the tag is in the distinct list of tags
         URI tagsUri = URI.create(jsonDatastoreConfig.url + "/tags/distinct");
         HttpRequest.Builder tagsBuilder = HttpRequest.newBuilder().uri(tagsUri);
-        HttpRequest tagsRequest = tagsBuilder
-                .header("Content-Type", "application/json")
-                .header("token", jsonDatastoreConfig.apiKey).build();
-        HttpResponse<String> response = client.send(tagsRequest, HttpResponse.BodyHandlers.ofString());
+
+        if (jsonDatastoreConfig.authentication instanceof APIKeyAuth) {
+            tagsBuilder
+                    .header("Content-Type", "application/json")
+                    .header("token", ((APIKeyAuth) jsonDatastoreConfig.authentication).apiKey);
+        } else {
+            tagsBuilder
+                    .header("Content-Type", "application/json");
+
+        }
```

Comment:
This could be simplified as we are setting `application/json` on both cases
```suggestion
        tagsBuilder
                .header("Content-Type", "application/json");
        if (jsonDatastoreConfig.authentication instanceof APIKeyAuth) {
            tagsBuilder
                    .header("token", ((APIKeyAuth) jsonDatastoreConfig.authentication).apiKey);
        }
```

Reply:
Done, thanks